### PR TITLE
Add tests for Staking Module

### PR DESCRIPTION
**Pull request by aries-kuchain**
staking add test file
handle_test
keeper/common_test
keeper/delegation_test
keeper/keeper_test
keeper/querier_test
keeper/slash_test
keeper/validator_test
fix some bug

### CODE REVIEW

File: x/staking/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+	"github.com/KuChainNetwork/kuchain/chain/config"
+	"github.com/KuChainNetwork/kuchain/chain/constants"
+	"github.com/KuChainNetwork/kuchain/chain/types"
+	"github.com/KuChainNetwork/kuchain/test/simapp"
+	stakingTypes "github.com/KuChainNetwork/kuchain/x/staking/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+func NewTestApp(wallet *simapp.Wallet) (addAlice, addJack, addValidator sdk.AccAddress, accAlice, accJack, accValidator types.AccountID, app *simapp.SimApp) {
```
Comment (aquarius-kuchain) [inline x/staking/handler_test.go]: Should newTestApp

File: x/staking/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+	So(genJack.GetAuth().Equals(accountJack.GetAuth()), ShouldBeTrue)
+
+	So(accountValidator, ShouldNotBeNil)
+	So(genValidator.GetID().Eq(accountValidator.GetID()), ShouldBeTrue)
+	So(genValidator.GetAuth().Equals(accountValidator.GetAuth()), ShouldBeTrue)
+
+	return addAlice, addJack, addValidator, accAlice, accJack, accValidator, app
+}
+
+func CreateValidator(t *testing.T, wallet *simapp.Wallet, app *simapp.SimApp, addAlice sdk.AccAddress, accAlice types.AccountID, rate sdk.Dec, pk crypto.PubKey, passed bool) error {
```
Comment (aquarius-kuchain) [inline x/staking/handler_test.go]: Should be createValudator if no use by others

File: x/staking/handler_test.go [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+}
+
+func TestStakingHandler(t *testing.T) {
+	Convey("TestValidatorHandler", t, func() {
+		config.SealChainConfig()
+		wallet := simapp.NewWallet()
+
+		addAlice, addJack, _, accAlice, accJack, _, app := NewTestApp(wallet)
+		wrongrate, _ := sdk.NewDecFromStr("1.1")
+		pk, _ := sdk.GetPubKeyFromBech32(sdk.Bech32PubKeyTypeConsPub, "kuchainvalconspub1zcjduepqn4usdx22zdntysj7n795xj77wrc62sytheeevr7zlna4yhwppdrs8mpds3")
```
Comment (aquarius-kuchain) [inline x/staking/handler_test.go]: use wallet to gen address

File: x/staking/keeper/keeper_test.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+func TestKeeper(t *testing.T) {
+	wallet := simapp.NewWallet()
+	Convey("TestParams", t, func() {
+		_, _, _, _, _, _, app := NewTestApp(wallet)
+		keeper := app.StakeKeeper()
+		keeper = keeper.EmptyHooks()
+		ctx := app.BaseApp.NewContext(true, abci.Header{Height: app.LastBlockHeight() + 1})
+		expParams := types.DefaultParams()
+		resParams := keeper.GetParams(ctx)
+		So(expParams.Equal(resParams), ShouldBeTrue)
```
Comment (aquarius-kuchain) [inline x/staking/keeper/keeper_test.go]: So(expParams, ShouldBeEqual, resParams)

or 

So(expParams, simapp.ShouldEq, resParams) for chain types

**Review by aquarius-kuchain — COMMENTED** [review]
Should fix some in review

File: x/staking/handler_test.go (hunk 1 above)
Comment (aries-kuchain) [reply x/staking/handler_test.go]: Modified

File: x/staking/handler_test.go (hunk 2 above)
Comment (aries-kuchain) [reply x/staking/handler_test.go]: Modified

File: x/staking/handler_test.go (hunk 3 above)
Comment (aries-kuchain) [reply x/staking/handler_test.go]: use function newPubKey to modify it

File: x/staking/keeper/keeper_test.go (hunk 4 above)
Comment (aries-kuchain) [reply x/staking/keeper/keeper_test.go]: use require.Equal

**Review by aquarius-kuchain — COMMENTED** [review]
OK

**Review by aquarius-kuchain — APPROVED** [review]

**Pull request merged**